src/modules/fancyzones/lib/WindowMoveHandler.cpp

```diff
@@ -0,0 +1,361 @@
+                if (zoneWindow && !moveSizeEnterCalled)
+                {
+                    zoneWindow->ShowZoneWindow();
+                }
+            }
+        }
+    }
+    else if (m_zoneWindowMoveSize)
+    {
+        m_zoneWindowMoveSize->RestoreOrginalTransparency();
+        m_zoneWindowMoveSize = nullptr;
+        for (auto [keyMonitor, zoneWindow] : zoneWindowMap)
+        {
+            if (zoneWindow)
+            {
+                zoneWindow->HideZoneWindow();
+            }
+        }
+    }
+}
+
+void WindowMoveHandlerPrivate::MoveSizeUpdate(HMONITOR monitor, POINT const& ptScreen, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept
+{
+    if (!m_inMoveSize)
+    {
+        return;
+    }
+
+    // This updates m_dragEnabled depending on if the shift key is being held down.
+    UpdateDragState(m_windowMoveSize);
+
+    if (m_zoneWindowMoveSize)
+    {
+        // Update the ZoneWindow already handling move/size
+        if (!m_dragEnabled)
+        {
+            // Drag got disabled, tell it to cancel and hide all windows
+            m_zoneWindowMoveSize = nullptr;
+
+            for (auto [keyMonitor, zoneWindow] : zoneWindowMap)
+            {
+                if (zoneWindow)
+                {
+                    zoneWindow->RestoreOrginalTransparency();
+                    zoneWindow->HideZoneWindow();
+                }
+            }
+        }
+        else
+        {
+            auto iter = zoneWindowMap.find(monitor);
+            if (iter != zoneWindowMap.end())
+            {
+                if (iter->second != m_zoneWindowMoveSize)
+                {
+                    // The drag has moved to a different monitor.
+                    m_zoneWindowMoveSize->RestoreOrginalTransparency();
+
+                    if (!m_settings->GetSettings()->showZonesOnAllMonitors)
+                    {
+                        m_zoneWindowMoveSize->HideZoneWindow();
+                    }
+                    m_zoneWindowMoveSize = iter->second;
+                    m_zoneWindowMoveSize->MoveSizeEnter(m_windowMoveSize, m_zoneWindowMoveSize->IsDragEnabled());
+                }
+
+                for (auto [keyMonitor, zoneWindow] : zoneWindowMap)
+                {
+                    zoneWindow->MoveSizeUpdate(ptScreen, m_dragEnabled);
+                }
+            }
+        }
+    }
+    else if (m_dragEnabled)
+    {
+        // We'll get here if the user presses/releases shift while dragging.
+        // Restart the drag on the ZoneWindow that m_windowMoveSize is on
+        MoveSizeStart(m_windowMoveSize, monitor, ptScreen, zoneWindowMap);
+        MoveSizeUpdate(monitor, ptScreen, zoneWindowMap);
+    }
+}
+
+void WindowMoveHandlerPrivate::MoveSizeEnd(HWND window, POINT const& ptScreen, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept
+{
+    if (window != m_windowMoveSize && !IsInterestingWindow(window, m_settings->GetSettings()->excludedAppsArray))

```

Comment:
this should be `||`, not `&&`

Reply:
You're right, thanks!

Reply:
Actually, to be completely consistent with the previous version, it should be &&, however, there shouldn't be a difference anyway since a window can't (?) change its "interestingness"

Reply:
It's the reverse of `if (window == m_windowMoveSize || IsInterestingWindow(window))` 
'&&' seems correct.